moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -567,72 +571,144 @@ TEST_F(OneRobot, FK)
-  EXPECT_EQ(traj.size(), 3);  // traj should be cut
-  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+  fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
+  EXPECT_EQ(expected_revolute_jump_traj_len, traj.size());  // traj should be cut
+  EXPECT_NEAR(expected_revolute_jump_fraction, fraction, 0.01);
 
-  // indirect call
+  // Indirect call using testJointSpaceJumps
   generateTestTraj(traj, robot_model, joint_model_group);
   fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(1.0, 1.0));
-  EXPECT_EQ(traj.size(), 3);  // traj should be cut
-  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+  EXPECT_EQ(expected_revolute_jump_traj_len, traj.size());  // traj should be cut
+  EXPECT_NEAR(expected_revolute_jump_fraction, fraction, 0.01);
 
-  // ignore revolute joints
+  // Ignore prismatic joints
+  generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(1.0, 0.0));
+  EXPECT_EQ(expected_revolute_jump_traj_len, traj.size());  // traj should be cut
+  EXPECT_NEAR(expected_revolute_jump_fraction, fraction, 0.01);
+
+  // Ignore revolute joints
   generateTestTraj(traj, robot_model, joint_model_group);
   fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(0.0, 1.0));
-  EXPECT_EQ(traj.size(), 4);  // traj should not be cut
-  EXPECT_NEAR(fraction, 4. / 4., 0.01);
+  EXPECT_EQ(expected_prismatic_jump_traj_len, traj.size());  // traj should be cut
+  EXPECT_NEAR(expected_prismatic_jump_fraction, fraction, 0.01);
+
+  // Ignore all absolute jumps
+  generateTestTraj(traj, robot_model, joint_model_group);
+  fraction = robot_state::RobotState::testJointSpaceJump(joint_model_group, traj, robot_state::JumpThreshold(0.0, 0.0));
+  EXPECT_EQ(expected_full_traj_len, traj.size());  // traj should be cut
+  EXPECT_NEAR(1.0, fraction, 0.01);
 }
 
 TEST_F(OneRobot, testRelativeJointSpaceJump)
 {
-  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_tip");
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_e");
   std::vector<std::shared_ptr<robot_state::RobotState>> traj;
 
-  // direct call of absolute version: factor slightly smaller than 3 (1.01 > 2.99 * 1.01 / 3)
+  // The first large jump of -1.01 occurs at the 5th waypoint so the test should trim the trajectory to lenght 4
+  const std::size_t expected_relative_jump_traj_len = 4;
+  const std::size_t expected_full_traj_len = 7;
+  const double expected_relative_jump_fraction =
+      (double)expected_relative_jump_traj_len / (double)expected_full_traj_len;
+
+  // Generate a test trajectory
+  std::size_t full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
```

Comment:
This was tested in `testAbsoluteJointSpaceJump()` already.

Reply:
Sound like a strange motivation to me: 
"Just in case somebody removes this line, I add backup one..."

Reply:
Good point, perhaps a better solution would be to separate this into it's own test

Reply:
Yes, that's an option.

Reply:
Sorry, obviously I haven't seen this question before. But I guess, I already complained about these global variables as the clutter the global scope and it's not obvious which tests actually use them.